Jellyfin.Api/Helpers/ProgressiveFileCopier.cs

```diff
@@ -48,37 +68,123 @@ public ProgressiveFileCopier(IStreamHelper streamHelper, IDirectStreamProvider d
         /// <returns>A <see cref="Task"/>.</returns>
         public async Task WriteToAsync(Stream outputStream, CancellationToken cancellationToken)
         {
-            if (_directStreamProvider != null)
+            cancellationToken = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, _cancellationToken).Token;
+
+            try
             {
-                await _directStreamProvider.CopyToAsync(outputStream, cancellationToken).ConfigureAwait(false);
-                return;
-            }
+                if (_directStreamProvider != null)
+                {
+                    await _directStreamProvider.CopyToAsync(outputStream, cancellationToken).ConfigureAwait(false);
+                    return;
+                }
+
+                var fileOptions = FileOptions.SequentialScan;
+                var allowAsyncFileRead = false;
+
+                // use non-async filestream along with read due to https://github.com/dotnet/corefx/issues/6039
+                if (Environment.OSVersion.Platform != PlatformID.Win32NT)
```

Comment:
Not sure if `RuntimeInformation.IsOSPlatform` is better here
https://docs.microsoft.com/en-us/dotnet/api/system.environment.osversion?view=netframework-4.7#remarks